chain/neutrino.go

```diff
@@ -369,6 +374,7 @@ func (s *NeutrinoClient) Rescan(startHash *chainhash.Hash, addrs []btcutil.Addre
 		case <-s.rescanQuit:
 			return nil
 		}
+		s.clientMtx.Lock()
```

Comment:
I think we must make sure the mutex is not double unlocked on quit here, since the defer will also unlock it.

Reply:
Fixed.